amqp/src/main/scala/akka/stream/alpakka/amqp/impl/AmqpPublishFlowStage.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp.impl
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.stream.alpakka.amqp.{AmqpPublishConfirmConfiguration, AmqpWriteSettings, WriteMessage}
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler}
+import akka.stream._
+
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/**
+ * Connects to an AMQP server upon materialization and sends incoming messages to the server.
+ * Each materialized flow will create one connection to the broker.
+ */
+@InternalApi
+private[amqp] final class AmqpPublishFlowStage[O](settings: AmqpWriteSettings)
+    extends GraphStageWithMaterializedValue[FlowShape[(WriteMessage, O), O], Future[Done]] { stage =>
+
+  val in = Inlet[(WriteMessage, O)]("AmqpPublishConfirmFlow.in")
+  val out = Outlet[O]("AmqpPublishConfirmFlow.out")
+
+  override def shape: FlowShape[(WriteMessage, O), O] = FlowShape.of(in, out)
+
+  override protected def initialAttributes: Attributes =
+    Attributes
+      .name("AmqpPublishFlowStage")
+      .and(ActorAttributes.dispatcher("akka.stream.default-blocking-io-dispatcher"))
+
+  override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, Future[Done]) = {
+    val promise = Promise[Done]()
```

Comment:
How would the materialized value be used? It seems just to get the stream failure or completion.

Reply:
Yes you're right, it doesn't seem to provide much value here. 
I've switched it to a `GraphStage`